### PR TITLE
Cql4bonnie multiple episode denex fixes

### PR DESCRIPTION
**Pull request by hossenlopp**
Fixes DENEX for multiple episode of care proportion measures to make it so any count in DENEX doesn't wipe out NUMER unless it is same or more than the count in DENOM.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1229
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint N/A

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] N/A JIRA tests have been run and pass
- [x] N/A You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Comment by hossenlopp** [comment]
Before:
----
Statements   : 56.92% ( 6736/11835 )
Branches     : 39.59% ( 1738/4390 )
Functions    : 47.46% ( 1122/2364 )
Lines        : 54.62% ( 6019/11019 )

After:
----
Statements   : 57% ( 6746/11835 )
Branches     : 39.81% ( 1748/4391 )
Functions    : 47.46% ( 1122/2364 )
Lines        : 54.71% ( 6029/11019 )

File: app/assets/javascripts/cql_calculator.js.coffee [hunk 1]
```diff
@@ -289,7 +289,7 @@
       resultShown.MSRPOPLEX = false if resultShown.MSRPOPLEX
 
     # If MSRPOPLEX is equal to MSRPOPL then OBSERVs are not calculated
```
Comment (lukoktonos) [inline app/assets/javascripts/cql_calculator.js.coffee]: Comment should be updated to say:
```
# If MSRPOPLEX is greater than or equal to MSRPOPL then OBSERVs are not calculated
```

File: app/assets/javascripts/cql_calculator.js.coffee [hunk 2]
```diff
@@ -194,8 +194,8 @@
         population_results['DENEXCEP'] = 0
       if 'NUMER' of population_results
         population_results['NUMER'] = 0
-    # Can not be in the numerator if excluded from the denominator
-    else if (population_results["DENEX"]? && !@isValueZero('DENEX', population_results))
+    # Can not be in the numerator if the same or more are excluded from the denominator
+    else if (population_results["DENEX"]? && !@isValueZero('DENEX', population_results) && (population_results["DENEX"] >= population_results["DENOM"]))
```
Comment (lukoktonos) [inline app/assets/javascripts/cql_calculator.js.coffee]: If `population_results["DENOM"] == 0` and `population_results["DENEX"] = 1`, then the else if statement on line 190 will get hit, which does not alter `'NUMEX'`.  Is that OK, or should `NUMEX` get 0ed out in that case?

**Review by lukoktonos — CHANGES_REQUESTED** [review]

**Review by lukoktonos — APPROVED** [review]

**Review by jbradl11 — APPROVED** [review]
Logic looks code. All tests pass.

**Pull request merged**